slsReceiverSoftware/src/ClientInterface.cpp

```diff
@@ -1427,4 +1405,23 @@ int ClientInterface::increment_file_index(Interface &socket) {
         impl()->setFileIndex(impl()->getFileIndex() + 1);
     }
     return socket.Send(OK);
-}
\ No newline at end of file
+}
+
+
+int ClientInterface::set_additional_json_parameter(Interface &socket) {
```

Comment:
Do we need special function for setting and getting? Couldn't we just put both key and value in one char array?

Reply:
I didnt really get you. The get function for json parameter is not special here as it uses only one char array. The set needs 2d char array. Using 1 d char array to include 2 strings is a bit ugly I feel, and also does not achieve any particular purpose. copyDetectorServer also uses 2 char arrays. Or did I misunderstand what you meant?